### ISSUE TITLE
Problem with image saving

### ISSUE THREAD

**Issue by JACKYNIKK** [openvinotoolkit/anomalib#534]
Sorry, i would like to save all the input images of my custom dataset on local disk in results folder. But if i select save images =True in config.yaml file only some random of them are saved in results. How must i do to save them all? Sorry for the silly question but has been 3 days trying and i can't find a solution
![immagine](https://user-images.githubusercontent.com/99801808/187938708-c54bc296-bcee-4dbd-8a61-2b79907cd9d8.png)

**Comment by yonafalinie** [comment]
Hi, as in my case I need to do the inference using _lightning_inference.py_ or _torch_inference.py_ in order to save the images.

**Comment by djdameln** [comment]
Hi, this is certainly not a silly question! The visualizer only saves generates images for the test set, so I'm guessing your folder of test/abnormal images is split between the validation set and the test set. Which dataset format are you using? And would you mind sharing the config file that you are using in your training run?

As suggested by @yonafalinie, an alternative approach would be to run the images through one of the inferencers.

**Comment by djdameln** [comment]
> If i understood your previous answer the images i saw in results folder when training where only the test images (which produced the metrics) ?

Yes, exactly. After a training run, we only visualize results for images in the test set. This is because the images in the validation set are not completely unseen by the model. These images might for example have been used to determine early stopping or to compute the adaptive threshold. As a result, the model might be biased towards these images, and the predictions on the validation set might not be a true reflection of the model's prediction capabilities. For this same reason, the images in the validation set are not included when computing the performance metrics.

For completeness, let me provide a bit more details on how the dataset splitting works in Anomalib:

The Folder datamodule automatically creates a random train/test split. Initially, the test set will include all images from the abnormal folder, as well as a random selection of normal images that are withheld from the training set. This is done so that images from the normal class are also represented in the evaluation process. The split ratio of the normal images can be configured using the `split_ratio` parameter (a `split_ratio` of 0.1 means that 10% of the normal images are withheld from training and included in testing).

When the `create_validation_set` parameter is set to `true`, the test set will additionally be split up between a test set and a validation set in a 50/50 ratio. Following the explanation above, it is recommended to use a validation set when your model makes use of early stopping, or when adaptive thresholding is enabled. When `create_validation_set` is set to `false`, the test set will serve a double purpose as the validation set.

So, in your case, since the `create_validation_set` parameter is enabled, the visualizations were only generated for 50% of the abnormal images that were randomly assigned to the test set. If you would disable the `create_validation_set` parameter, the test set would contain all abnormal images in the dataset and these would all be visualized. Keep in mind however, that the performance numbers and predictions might overestimate the true model performance in that case, because the model will already have seen the images in the validation stage.

**Comment by JACKYNIKK** [comment]
Thank you so much for this exhaustive explanation. I'm closing the issue since my doubt has been solve. Thank you again
